staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go

```diff
@@ -186,98 +196,153 @@ func validateStorageVersion(t *testing.T, ctc *conversionTestContext) {
 
 // validateMixedStorageVersions ensures that identical custom resources written at different storage versions
 // are readable and remain the same.
-func validateMixedStorageVersions(t *testing.T, ctc *conversionTestContext) {
-	ns := ctc.namespace
+func validateMixedStorageVersions(versions ...string) func(t *testing.T, ctc *conversionTestContext) {
+	return func(t *testing.T, ctc *conversionTestContext) {
+		ns := ctc.namespace
+		clients := ctc.versionedClients(ns)
 
-	v1client := ctc.versionedClient(ns, "v1beta1")
-	v2client := ctc.versionedClient(ns, "v1beta2")
-	clients := map[string]dynamic.ResourceInterface{"v1beta1": v1client, "v1beta2": v2client}
-	versions := []string{"v1beta1", "v1beta2"}
+		// Create CRs at all storage versions
+		objNames := []string{}
+		for _, version := range versions {
+			ctc.setAndWaitStorageVersion(t, version)
 
-	// Create CRs at all storage versions
-	objNames := []string{}
-	for _, version := range versions {
-		ctc.setAndWaitStorageVersion(t, version)
+			name := "mixedstorage-stored-as-" + version
+			obj, err := clients[version].Create(newConversionMultiVersionFixture(ns, name, version), metav1.CreateOptions{})
+			if err != nil {
+				t.Fatal(err)
+			}
+			objNames = append(objNames, obj.GetName())
+		}
 
-		name := "stored-at-" + version
-		obj, err := clients[version].Create(newConversionMultiVersionFixture(ns, name, version), metav1.CreateOptions{})
-		if err != nil {
-			t.Fatal(err)
+		// Ensure copies of an object have the same fields and values at each custom resource definition version regardless of storage version
+		for clientVersion, client := range clients {
+			t.Run(clientVersion, func(t *testing.T) {
```

Comment:
naming the subtests `mixed-<version>` or `served-<version>`, etc, would make it possible to tell which test failed by the name. would suggest tweaking this in a follow-up, this PR looks reasonable to merge as is and would like to unblock.

Reply:
yes, was near to change that as well.